website/src/controllers/api-v3/groups.js

```diff
@@ -295,6 +295,12 @@ api.joinGroup = {
 
     if (group.type === 'party' && inviter) {
       promises.push(User.update({_id: inviter}, {$inc: {'items.quests.basilist': 1}}).exec()); // Reward inviter
+      if (group.memberCount > 1) {
+        promises.push(User.update({$or: [{'party._id': group._id}, {_id: user._id}], 'achievements.partyUp': {$nin: [true]}}, {$set: {'achievements.partyUp': true}}, {multi: true}).exec());
```

Comment:
instead of $nin I think here you could use $ne https://docs.mongodb.org/manual/reference/operator/query/ne/